cdk/src/open-data-platform/data-plane/data-import/handler-factory.ts

```diff
@@ -1,108 +1,199 @@
 import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
-import { ConnectionPool } from '@databases/pg';
 import * as AWS from 'aws-sdk';
-import { connectToDb } from '../schema/schema.handler';
+import { AnyLengthString } from 'aws-sdk/clients/comprehendmedical';
 
 // These libraries don't have types, so they are imported in a different way.
 const { chain } = require('stream-chain');
 const { streamArray } = require('stream-json/streamers/StreamArray');
 const Batch = require('stream-json/utils/Batch');
 const Pick = require('stream-json/filters/Pick');
 
+export interface ImportRequest {
+  rowOffset?: number;
+  rowLimit?: number;
+  batchSize?: number;
+}
+
+interface ImportResult {
+  processedBatchCount: number;
+  sucessfulBatchCount: number;
+  erroredBatchCount: number;
+}
+
+// These are defined here because the geoJsonHandlerFactory function signature was unreadable.
+type ImportCallback = (rows: any[], db: AWS.RDSDataService) => Promise<void>;
+type ImportHandler = (event: ImportRequest) => Promise<APIGatewayProxyResult>;
+
 const S3 = new AWS.S3();
 
 /**
  * Builds a lambda handler that parses a GeoJSON file and executes a callback.
+ *
+ * The callback takes in a set of rows and should insert them into the DB using the provided db
```

Comment:
nit: even though this is currently only used for insert, nothing about this is specific to inserting right? if not maybe we should keep documentation general - i.e. "The callback takes in a set of rows and a connection to the db and can do anything with those rows"... maybe not exactly that wording but you get the gist!

Reply:
I think that's a fair point, but I kept it as-is here since the only use for the callback right now is inserts. So it tells the user what they should put in the callback. But I'm up to change it if we ever use this for another purpose.